### PR TITLE
Backport changes from mozilla-central

### PR DESCRIPTION
**Pull request by mythmon**
This backports a patch @gijsk made to mozilla-central in bug 1330954, and adds a scripts to make backporting the tiniest bit easier.

### CODE REVIEW

File: recipe-client-addon/lib/Heartbeat.jsm [hunk 1]
```diff
@@ -315,12 +313,16 @@ this.Heartbeat = class {
       this.chromeWindow.gBrowser.selectedTab = this.chromeWindow.gBrowser.addTab(this.options.postAnswerUrl.toString());
     }
 
-    if (this.surveyEndTimer) {
-      clearTimeout(this.surveyEndTimer);
-      this.surveyEndTimer = null;
-    }
+    this.endTimerIfPresent("surveyEndTimer");
+
+    this.engagementCloseTimer = setTimeout(() => this.close(), NOTIFICATION_TIME);
```
Comment (Osmose) [inline recipe-client-addon/lib/Heartbeat.jsm]: @gijsk Why use the callback form of this if `this.close` is already bound?

Comment (gijsk) [inline recipe-client-addon/lib/Heartbeat.jsm]: The binding thing happened in a patch revision (cf. last point of https://bugzilla.mozilla.org/show_bug.cgi?id=1330954#c2 ) and I forgot to update this - good find! We could indeed just pass this.close directly.

**Comment by gijsk** [comment]
Also, sorry about doing this on bugzilla + m-c -- in hindsight, I should have used github. :-(

File: recipe-client-addon/lib/Heartbeat.jsm (hunk 1 above)
Comment (Osmose) [inline recipe-client-addon/lib/Heartbeat.jsm]: Cool, we'll merge this and then fix the nit in another PR to help track the backbackport. :P

**Pull request merged**